### PR TITLE
Twos complement

### PR DESCRIPTION
**Pull request by rnugent3**
sorry this is so much in one branch

### CODE REVIEW

File: fda-model/metrics/ImpactAreaScenarioResults.cs [hunk 1]
```diff
@@ -168,10 +171,16 @@ public bool ResultsAreConverged(double upperConfidenceLimitProb, double lowerCon
...
-                    {
-                        if (consequenceDistributionResult.ConsequenceHistogram.IsHistogramConverged(upperConfidenceLimitProb, lowerConfidenceLimitProb) == false)
+                    {   
+                        if(consequenceDistributionResult.ConsequenceHistogram.HistogramIsZeroValued)
+                        {
+                            eadIsConverged = true;
+                        } else
                         {
-                            eadIsConverged = false;
+                            if (consequenceDistributionResult.ConsequenceHistogram.IsHistogramConverged(upperConfidenceLimitProb, lowerConfidenceLimitProb) == false)
```
Comment (codymccoy) [inline fda-model/metrics/ImpactAreaScenarioResults.cs:180]: This is fine, and is maybe easier to read, but often if you are doing a check and then the only thing you are doing in the check is assigning a boolean value, you can just do that in one line:

eadIsConverged = consequenceDistributionResult.ConsequenceHistogram.IsHistogramConverged(upperConfidenceLimitProb, lowerConfidenceLimitProb);

File: fda-model/paireddata/GraphicalUncertainPairedData.cs [hunk 2]
```diff
@@ -23,10 +23,17 @@ public class GraphicalUncertainPairedData : HEC.MVVMFramework.Base.Implementatio
         private bool _UsingStagesNotFlows;
         private double _MaximumProbability;
         private double _MinimumProbability;
-
+        private double[] _inputFlowOrStageValues;
```
Comment (codymccoy) [inline fda-model/paireddata/GraphicalUncertainPairedData.cs:26]: We have been capitalizing the first letter.

File: fda-model/alternatives/Alternative.cs [hunk 3]
```diff
@@ -32,41 +28,109 @@ public static AlternativeResults AnnualizationCompute(interfaces.IProvideRandomN
...
+            {
+                futureYearResultsList.Add(futureYearImpactAreaScenarioResults);
+            }
+
             foreach (ImpactAreaScenarioResults baseYearResults in computedResultsBaseYear.ResultsList)
             {
                 ImpactAreaScenarioResults mlfYearResults = computedResultsFutureYear.GetResults(baseYearResults.ImpactAreaID);
+                futureYearResultsList.Remove(mlfYearResults);
+
+                List<ConsequenceDistributionResult> mlfYearDamageResultsList = new List<ConsequenceDistributionResult>();
```
[the remark this answers is not in the document]
Comment (rnugent3) [reply fda-model/alternatives/Alternative.cs:42]: true, but I don't think we know the blah blah blahs to do that

File: fda-model/compute/ImpactAreaScenarioSimulation.cs [hunk 4]
```diff
@@ -648,6 +649,171 @@ private void TopOfLeveehasCertainFailure()
                 ReportMessage(this, new MessageEventArgs(errorMessage));
             }
         }
+        //TODO: Add messaging to indicate which curves do not overlap
+        private bool SimulationCurvesHaveOverlap()
+        {
+            bool allCurvesHaveOverlap = true;
```
[the remark this answers is not in the document]
Comment (rnugent3) [reply fda-model/compute/ImpactAreaScenarioSimulation.cs:655]: to make matters worse, this sequence of if statements is repeated several times in this class. there is definitely an opportunity for refactoring.

File: fda-model/compute/ImpactAreaScenarioSimulation.cs [hunk 5]
```diff
@@ -648,6 +649,171 @@ private void TopOfLeveehasCertainFailure()
...
+                    {
+                        bool stageDamageOverlaps = CurvesHaveOverlap(uncertain, _frequency_stage);
+                        if (!stageDamageOverlaps)
+                        {
+                            allCurvesHaveOverlap = stageDamageOverlaps;
+                        }
+                    }
+                }
+            }
+            return allCurvesHaveOverlap;
```
[the remark this answers is not in the document]
Comment (rnugent3) [reply fda-model/compute/ImpactAreaScenarioSimulation.cs:764]: would you believe that I already did refactor 1 massive method into 4 methods. further refactoring is definitely warranted

**Pull request merged**